agent/connect/ca/provider_vault.go

```diff
@@ -396,17 +396,14 @@ func (v *VaultProvider) SetIntermediate(intermediatePEM, rootPEM string) error {
 		return fmt.Errorf("cannot set an intermediate using another root in the primary datacenter")
 	}
 
-	err := validateSetIntermediate(
-		intermediatePEM, rootPEM,
-		"", // we don't have access to the private key directly
-		v.spiffeID,
-	)
+	// the private key is in vault, so we can't use it in this validation
+	err := validateSetIntermediate(intermediatePEM, rootPEM, "", v.spiffeID)
 	if err != nil {
 		return err
 	}
 
 	_, err = v.client.Logical().Write(v.config.IntermediatePKIPath+"intermediate/set-signed", map[string]interface{}{
-		"certificate": fmt.Sprintf("%s\n%s", intermediatePEM, rootPEM),
+		"certificate": intermediatePEM,
```

Comment:
The `intermediatePEM` already has the root cert appended. I can't see anywhere we do this in Consul, so it must come from the CSR response returned by Vault. The Vault API docs are not really clear about this: https://www.vaultproject.io/api/secret/pki

Reply:
Some questions about this:
1. Do other providers also append the root cert to the PEM when responding to CSR requests? From what I can tell, the Consul provider does not.
2. Does it matter? Maybe having the root cert PEM here is only important for Vault.

Reply:
OK, so looking at other providers:

* The consul provider [stores it](https://github.com/hashicorp/consul/blob/main/agent/connect/ca/provider_consul.go#L268), but [never uses it](https://github.com/hashicorp/consul/blob/main/agent/connect/ca/provider_consul.go#L280-L290)  (because `SetIntermediate` is only called in secondary DCs). 
* The aws provider does basically the same thing as the Consul provider. Stores it in-process, but never uses it in the secondary.

So I think this change is only relevant to the vault provider.